looking through the job output the problem seems to rely roughly here:

```
[2021-11-12T10:55:07.872Z] + /srv/builder/bldr switch_revision_update_instance:journal--prod,fc9a1b2f25bb7f5dbc7e841d6389aafb001d0e88,concurrency=blue-green
[2021-11-12T10:55:08.123Z] /ext/srv/builder/venv/local/lib/python2.7/site-packages/gevent/builtins.py:96: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
[2021-11-12T10:55:08.123Z]   result = _import(*args, **kwargs)
[2021-11-12T10:55:10.606Z] /ext/srv/builder/venv/local/lib/python2.7/site-packages/boto3/compat.py:86: PythonDeprecationWarning: Boto3 will no longer support Python 2.7 starting July 15, 2021. To continue receiving service updates, bug fixes, and security updates please upgrade to Python 3.6 or later. More information can be found here: https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-python-2-7-in-aws-sdk-for-python-and-aws-cli-v1/
[2021-11-12T10:55:10.606Z]   warnings.warn(warning, PythonDeprecationWarning)
[2021-11-12T10:55:10.606Z] INFO - buildercore.core - private key for the bootstrap user for this host is not present locally (/ext/srv/builder/.cfn/keypairs/journal--prod.pem); will not override ~/.ssh with it.
[2021-11-12T10:55:10.606Z] INFO - buildercore.core - Executing on ec2 nodes ({'i-0440759a13f7f2c0d': '3.91.65.159', 'i-09d3991bf9232ddcc': '3.93.68.16', 'i-0224bf77e65c609de': '54.81.74.129'}), concurrency None
[2021-11-12T10:55:10.606Z] [3.91.65.159] Executing task 'single_node_work_fn'
[2021-11-12T10:55:10.606Z] [3.93.68.16] Executing task 'single_node_work_fn'
[2021-11-12T10:55:10.606Z] [54.81.74.129] Executing task 'single_node_work_fn'
[2021-11-12T10:55:12.468Z] INFO - buildvars - updating 'journal--prod' with new vars {u'rds_dbname': None, u'domain': u'elifesciences.org', u'int_project_hostname': None, u'full_hostname': u'prod--journal.elifesciences.org', u'stackname': u'journal--prod', u'cluster_id': None, u'ec2': {u'cluster-size': 3, u'ami': u'ami-0d65d77807d45828c', u'dns-internal': False, u'dns-external-primary': False, u'type': u't2.small', u'overrides': {}, u'masterless': False, u'suppressed': [], u'security-group': {}, u'master_ip': u'10.0.2.135', u'cpu-credits': u'standard', u'ports': [22, 443, {u'80': {u'cidr-ip': u'10.0.0.0/16'}}]}, u'subdomains': [], u'instance_id': u'prod', u'rds_instance_id': None, u'author': u'elife', u's3': {}, u'hostname': u'prod--journal', u'date_rendered': u'2021-06-07', u'gcs': False, u'int_full_hostname': None, u'subdomain': u'journal', u'alt-config': u'prod', u'revision': 'fc9a1b2f25bb7f5dbc7e841d6389aafb001d0e88', u'int_domain': None, u'project_name': u'journal', u'node': 1, u'nodename': u'journal--prod--1', u'sqs': {}, u'fastly': {u'errors': {u'url': u'https://prod-elife-error-pages.s3.amazonaws.com/', u'codes': {u'404': u'404.html', u'410': u'410.html', u'503': u'503.html'}, u'fallbacks': {u'5xx': u'5xx.html', u'4xx': u'4xx.html'}}, u'vcl-templates': {}, u'shield': {u'pop': u'dca-dc-us'}, u'healthcheck': {u'path': u'/ping-fastly', u'check-interval': 10000, u'timeout': 2000}, u'vcl': [u'original-host', u'ping-status', u'strip-non-journal-cookies', u'journal-google-scholar', u'journal-google-scholar-vary'], u'bigquerylogging': {u'project': u'elife-fastly', u'table': u'journal', u'dataset': u'prod'}, u'subdomains': [u'prod--cdn-journal.elifesciences.org', u'placeholder-prod-journal.elifesciences.org', u'elifesciences.org'], u'surrogate-keys': {}, u'backends': {}, u'dns': {u'a': [u'151.101.2.217', u'151.101.66.217', u'151.101.130.217', u'151.101.194.217'], u'cname': u'u2.shared.global.fastly.net'}, u'ip-blacklist': True, u'subdomains-without-dns': [], u'gcslogging': {u'path': u'journal--prod/', u'bucket': u'prod-elife-fastly', u'period': 600}, u'default-ttl': 3600}, u'aws': {u'account-id': 512686554592, u'region': u'us-east-1', u'redundant-subnet-cidr': u'10.0.3.0/24', u'redundant-subnet-id': u'subnet-7a31dd46', u'subnet-cidr': u'10.0.2.0/24', u'subnet-id': u'subnet-1d4eb46a', u'redundant-availability-zone': u'us-east-1e', u'availability-zone': u'us-east-1d', u'vpc-id': u'vpc-78a2071d'}, u'cloudfront': {u'cookies': [], u'errors': {u'pattern': u'???.html', u'domain': u'prod-elife-error-pages.s3-website-us-east-1.amazonaws.com', u'codes': {u'503': u'/5xx.html'}, u'protocol': u'http'}, u'logging': False, u'origins': {}, u'certificate_id': u'ASCAXOXT77XQNRYWRB7BT', u'compress': True, u'subdomains': [u'placeholder2-prod-journal.elifesciences.org', u'www.elifesciences.org', u'elife.elifesciences.org', u'prod.elifesciences.org'], u'headers': [u'Host'], u'subdomains-without-dns': [], u'default-ttl': 0}, u'sns': [], u'branch': u'master', u'vault': {u'address': u'https://master-server.elifesciences.org:8200'}, u'rds': {}, u'docdb': False, u'ext_node_hostname': u'prod--journal--%s.elifesciences.org', u'bigquery': False, u'rds_password': None, u'project': {u'formula-repo': u'https://github.com/elifesciences/journal-formula', u'formula-dependencies': [u'https://github.com/elifesciences/builder-base-formula', u'https://github.com/elifesciences/api-dummy-formula']}, u'elb': {u'subnets': [u'subnet-1d4eb46a', u'subnet-7a31dd46'], u'protocol': [u'https', u'http'], u'certificate': u'arn:aws:iam::512686554592:server-certificate/cloudfront/wildcard.elifesciences.org/2021-02-11.wildcard.elifesciences.org', u'healthcheck': {u'protocol': u'http', u'healthy_threshold': 2, u'interval': 5, u'unhealthy_threshold': 2, u'timeout': 4, u'path': u'/ping', u'port': 80}, u'idle_timeout': 60, u'additional_listeners': {}, u'stickiness': False}, u'ext': False, u'project_hostname': u'journal.elifesciences.org', u'rds_username': None, u'elasticache': {u'engine': u'redis', u'subnets': [u'subnet-20275c68', u'subnet-c4033af8'], u'az': u'us-east-1d', u'overrides': {u'2': {u'type': u'cache.t2.micro'}}, u'version': u'2.8.24', u'clusters': 2, u'configuration': {u'maxmemory-policy': u'volatile-lru'}, u'type': u'cache.t2.medium'}, u'eks': False}
[2021-11-12T10:55:12.468Z] INFO - buildvars - updating 'journal--prod' with new vars {u'rds_dbname': None, u'domain': u'elifesciences.org', u'int_project_hostname': None, u'full_hostname': u'prod--journal.elifesciences.org', u'stackname': u'journal--prod', u'cluster_id': None, u'ec2': {u'cluster-size': 3, u'ami': u'ami-0d65d77807d45828c', u'dns-internal': False, u'dns-external-primary': False, u'type': u't2.small', u'overrides': {}, u'masterless': False, u'suppressed': [], u'security-group': {}, u'master_ip': u'10.0.2.135', u'cpu-credits': u'standard', u'ports': [22, 443, {u'80': {u'cidr-ip': u'10.0.0.0/16'}}]}, u'subdomains': [], u'instance_id': u'prod', u'rds_instance_id': None, u'author': u'elife', u's3': {}, u'hostname': u'prod--journal', u'date_rendered': u'2021-06-07', u'gcs': False, u'int_full_hostname': None, u'subdomain': u'journal', u'alt-config': u'prod', u'revision': 'fc9a1b2f25bb7f5dbc7e841d6389aafb001d0e88', u'int_domain': None, u'project_name': u'journal', u'node': 3, u'nodename': u'journal--prod--3', u'sqs': {}, u'fastly': {u'errors': {u'url': u'https://prod-elife-error-pages.s3.amazonaws.com/', u'codes': {u'404': u'404.html', u'410': u'410.html', u'503': u'503.html'}, u'fallbacks': {u'5xx': u'5xx.html', u'4xx': u'4xx.html'}}, u'vcl-templates': {}, u'shield': {u'pop': u'dca-dc-us'}, u'healthcheck': {u'path': u'/ping-fastly', u'check-interval': 10000, u'timeout': 2000}, u'vcl': [u'original-host', u'ping-status', u'strip-non-journal-cookies', u'journal-google-scholar', u'journal-google-scholar-vary'], u'bigquerylogging': {u'project': u'elife-fastly', u'table': u'journal', u'dataset': u'prod'}, u'subdomains': [u'prod--cdn-journal.elifesciences.org', u'placeholder-prod-journal.elifesciences.org', u'elifesciences.org'], u'surrogate-keys': {}, u'backends': {}, u'dns': {u'a': [u'151.101.2.217', u'151.101.66.217', u'151.101.130.217', u'151.101.194.217'], u'cname': u'u2.shared.global.fastly.net'}, u'ip-blacklist': True, u'subdomains-without-dns': [], u'gcslogging': {u'path': u'journal--prod/', u'bucket': u'prod-elife-fastly', u'period': 600}, u'default-ttl': 3600}, u'aws': {u'account-id': 512686554592, u'region': u'us-east-1', u'redundant-subnet-cidr': u'10.0.3.0/24', u'redundant-subnet-id': u'subnet-7a31dd46', u'subnet-cidr': u'10.0.2.0/24', u'subnet-id': u'subnet-1d4eb46a', u'redundant-availability-zone': u'us-east-1e', u'availability-zone': u'us-east-1d', u'vpc-id': u'vpc-78a2071d'}, u'cloudfront': {u'cookies': [], u'errors': {u'pattern': u'???.html', u'domain': u'prod-elife-error-pages.s3-website-us-east-1.amazonaws.com', u'codes': {u'503': u'/5xx.html'}, u'protocol': u'http'}, u'logging': False, u'origins': {}, u'certificate_id': u'ASCAXOXT77XQNRYWRB7BT', u'compress': True, u'subdomains': [u'placeholder2-prod-journal.elifesciences.org', u'www.elifesciences.org', u'elife.elifesciences.org', u'prod.elifesciences.org'], u'headers': [u'Host'], u'subdomains-without-dns': [], u'default-ttl': 0}, u'sns': [], u'branch': u'master', u'vault': {u'address': u'https://master-server.elifesciences.org:8200'}, u'rds': {}, u'docdb': False, u'ext_node_hostname': u'prod--journal--%s.elifesciences.org', u'bigquery': False, u'rds_password': None, u'project': {u'formula-repo': u'https://github.com/elifesciences/journal-formula', u'formula-dependencies': [u'https://github.com/elifesciences/builder-base-formula', u'https://github.com/elifesciences/api-dummy-formula']}, u'elb': {u'subnets': [u'subnet-1d4eb46a', u'subnet-7a31dd46'], u'protocol': [u'https', u'http'], u'certificate': u'arn:aws:iam::512686554592:server-certificate/cloudfront/wildcard.elifesciences.org/2021-02-11.wildcard.elifesciences.org', u'healthcheck': {u'protocol': u'http', u'healthy_threshold': 2, u'interval': 5, u'unhealthy_threshold': 2, u'timeout': 4, u'path': u'/ping', u'port': 80}, u'idle_timeout': 60, u'additional_listeners': {}, u'stickiness': False}, u'ext': False, u'project_hostname': u'journal.elifesciences.org', u'rds_username': None, u'elasticache': {u'engine': u'redis', u'subnets': [u'subnet-20275c68', u'subnet-c4033af8'], u'az': u'us-east-1d', u'overrides': {u'2': {u'type': u'cache.t2.micro'}}, u'version': u'2.8.24', u'clusters': 2, u'configuration': {u'maxmemory-policy': u'volatile-lru'}, u'type': u'cache.t2.medium'}, u'eks': False}
[2021-11-12T10:55:12.469Z] INFO - buildvars - updating 'journal--prod' with new vars {u'rds_dbname': None, u'domain': u'elifesciences.org', u'int_project_hostname': None, u'full_hostname': u'prod--journal.elifesciences.org', u'stackname': u'journal--prod', u'cluster_id': None, u'ec2': {u'cluster-size': 3, u'ami': u'ami-0d65d77807d45828c', u'dns-internal': False, u'dns-external-primary': False, u'type': u't2.small', u'overrides': {}, u'masterless': False, u'suppressed': [], u'security-group': {}, u'master_ip': u'10.0.2.135', u'cpu-credits': u'standard', u'ports': [22, 443, {u'80': {u'cidr-ip': u'10.0.0.0/16'}}]}, u'subdomains': [], u'instance_id': u'prod', u'rds_instance_id': None, u'author': u'elife', u's3': {}, u'hostname': u'prod--journal', u'date_rendered': u'2021-06-07', u'gcs': False, u'int_full_hostname': None, u'subdomain': u'journal', u'alt-config': u'prod', u'revision': 'fc9a1b2f25bb7f5dbc7e841d6389aafb001d0e88', u'int_domain': None, u'project_name': u'journal', u'node': 2, u'nodename': u'journal--prod--2', u'sqs': {}, u'fastly': {u'errors': {u'url': u'https://prod-elife-error-pages.s3.amazonaws.com/', u'codes': {u'404': u'404.html', u'410': u'410.html', u'503': u'503.html'}, u'fallbacks': {u'5xx': u'5xx.html', u'4xx': u'4xx.html'}}, u'vcl-templates': {}, u'shield': {u'pop': u'dca-dc-us'}, u'healthcheck': {u'path': u'/ping-fastly', u'check-interval': 10000, u'timeout': 2000}, u'vcl': [u'original-host', u'ping-status', u'strip-non-journal-cookies', u'journal-google-scholar', u'journal-google-scholar-vary'], u'bigquerylogging': {u'project': u'elife-fastly', u'table': u'journal', u'dataset': u'prod'}, u'subdomains': [u'prod--cdn-journal.elifesciences.org', u'placeholder-prod-journal.elifesciences.org', u'elifesciences.org'], u'surrogate-keys': {}, u'backends': {}, u'dns': {u'a': [u'151.101.2.217', u'151.101.66.217', u'151.101.130.217', u'151.101.194.217'], u'cname': u'u2.shared.global.fastly.net'}, u'ip-blacklist': True, u'subdomains-without-dns': [], u'gcslogging': {u'path': u'journal--prod/', u'bucket': u'prod-elife-fastly', u'period': 600}, u'default-ttl': 3600}, u'aws': {u'account-id': 512686554592, u'region': u'us-east-1', u'redundant-subnet-cidr': u'10.0.3.0/24', u'redundant-subnet-id': u'subnet-7a31dd46', u'subnet-cidr': u'10.0.2.0/24', u'subnet-id': u'subnet-1d4eb46a', u'redundant-availability-zone': u'us-east-1e', u'availability-zone': u'us-east-1d', u'vpc-id': u'vpc-78a2071d'}, u'cloudfront': {u'cookies': [], u'errors': {u'pattern': u'???.html', u'domain': u'prod-elife-error-pages.s3-website-us-east-1.amazonaws.com', u'codes': {u'503': u'/5xx.html'}, u'protocol': u'http'}, u'logging': False, u'origins': {}, u'certificate_id': u'ASCAXOXT77XQNRYWRB7BT', u'compress': True, u'subdomains': [u'placeholder2-prod-journal.elifesciences.org', u'www.elifesciences.org', u'elife.elifesciences.org', u'prod.elifesciences.org'], u'headers': [u'Host'], u'subdomains-without-dns': [], u'default-ttl': 0}, u'sns': [], u'branch': u'master', u'vault': {u'address': u'https://master-server.elifesciences.org:8200'}, u'rds': {}, u'docdb': False, u'ext_node_hostname': u'prod--journal--%s.elifesciences.org', u'bigquery': False, u'rds_password': None, u'project': {u'formula-repo': u'https://github.com/elifesciences/journal-formula', u'formula-dependencies': [u'https://github.com/elifesciences/builder-base-formula', u'https://github.com/elifesciences/api-dummy-formula']}, u'elb': {u'subnets': [u'subnet-1d4eb46a', u'subnet-7a31dd46'], u'protocol': [u'https', u'http'], u'certificate': u'arn:aws:iam::512686554592:server-certificate/cloudfront/wildcard.elifesciences.org/2021-02-11.wildcard.elifesciences.org', u'healthcheck': {u'protocol': u'http', u'healthy_threshold': 2, u'interval': 5, u'unhealthy_threshold': 2, u'timeout': 4, u'path': u'/ping', u'port': 80}, u'idle_timeout': 60, u'additional_listeners': {}, u'stickiness': False}, u'ext': False, u'project_hostname': u'journal.elifesciences.org', u'rds_username': None, u'elasticache': {u'engine': u'redis', u'subnets': [u'subnet-20275c68', u'subnet-c4033af8'], u'az': u'us-east-1d', u'overrides': {u'2': {u'type': u'cache.t2.micro'}}, u'version': u'2.8.24', u'clusters': 2, u'configuration': {u'maxmemory-policy': u'volatile-lru'}, u'type': u'cache.t2.medium'}, u'eks': False}
[2021-11-12T10:55:12.469Z] INFO - buildvars - 'journal--prod' updated. backup written to /tmp/build-vars.json.b64.20211112105512
[2021-11-12T10:55:12.469Z] [3.93.68.16] download: <file obj> <- /etc/build-vars.json.b64
[2021-11-12T10:55:12.469Z] [3.93.68.16] sudo: if [ -f /etc/build-vars.json.b64 ]; then cp /etc/build-vars.json.b64 /tmp/build-vars.json.b64.20211112105512; fi;
[2021-11-12T10:55:12.469Z] [3.93.68.16] put: <file obj> -> /etc/build-vars.json.b64
[2021-11-12T10:55:12.469Z] INFO - buildvars - 'journal--prod' updated. backup written to /tmp/build-vars.json.b64.20211112105512
[2021-11-12T10:55:12.469Z] [54.81.74.129] download: <file obj> <- /etc/build-vars.json.b64
[2021-11-12T10:55:12.469Z] [54.81.74.129] sudo: if [ -f /etc/build-vars.json.b64 ]; then cp /etc/build-vars.json.b64 /tmp/build-vars.json.b64.20211112105512; fi;
[2021-11-12T10:55:12.469Z] [54.81.74.129] put: <file obj> -> /etc/build-vars.json.b64
[2021-11-12T10:55:12.469Z] INFO - buildvars - 'journal--prod' updated. backup written to /tmp/build-vars.json.b64.20211112105512
[2021-11-12T10:55:12.469Z] [3.91.65.159] download: <file obj> <- /etc/build-vars.json.b64
[2021-11-12T10:55:12.469Z] [3.91.65.159] sudo: if [ -f /etc/build-vars.json.b64 ]; then cp /etc/build-vars.json.b64 /tmp/build-vars.json.b64.20211112105512; fi;
[2021-11-12T10:55:12.469Z] [3.91.65.159] put: <file obj> -> /etc/build-vars.json.b64
[2021-11-12T10:55:12.719Z] INFO - buildercore.s3 - downloading key contexts/journal--prod.json
[2021-11-12T10:55:12.970Z] INFO - buildercore.s3 - downloading key contexts/journal--prod.json
[2021-11-12T10:55:14.313Z] INFO - buildercore.core - private key for the bootstrap user for this host is not present locally (/ext/srv/builder/.cfn/keypairs/journal--prod.pem); will not override ~/.ssh with it.
[2021-11-12T10:55:14.313Z] INFO - buildercore.core - Executing on ec2 nodes ({'i-0440759a13f7f2c0d': '3.91.65.159', 'i-09d3991bf9232ddcc': '3.93.68.16', 'i-0224bf77e65c609de': '54.81.74.129'}), concurrency <function do at 0x7fd682dae5d0>
[2021-11-12T10:55:14.313Z] INFO - buildercore.utils - Waiting for all instances to be in service...
[2021-11-12T10:55:15.657Z] INFO - buildercore.bluegreen_v2 - Instance statuses on arn:aws:elasticloadbalancing:us-east-1:512686554592:loadbalancer/app/journal--prod/677b5066e869788d: {'i-0440759a13f7f2c0d': 'healthy', 'i-09d3991bf9232ddcc': 'healthy', 'i-0224bf77e65c609de': 'healthy'}
[2021-11-12T10:55:15.657Z] INFO - buildercore.utils - done.
[2021-11-12T10:55:16.208Z] INFO - buildercore.bluegreen_v2 - Blue phase on arn:aws:elasticloadbalancing:us-east-1:512686554592:loadbalancer/app/journal--prod/677b5066e869788d: i-0440759a13f7f2c0d, i-09d3991bf9232ddcc
[2021-11-12T10:55:16.759Z] INFO - buildercore.bluegreen_v2 - deregistering targets: [{'Id': 'i-0440759a13f7f2c0d'}, {'Id': 'i-09d3991bf9232ddcc'}]
[2021-11-12T10:55:17.310Z] INFO - buildercore.bluegreen_v2 - Waiting for deregistration of all on arn:aws:elasticloadbalancing:us-east-1:512686554592:loadbalancer/app/journal--prod/677b5066e869788d: i-0440759a13f7f2c0d, i-09d3991bf9232ddcc
[2021-11-12T10:55:17.310Z] INFO - buildercore.utils - waiting ...
[2021-11-12T10:55:18.653Z] INFO - buildercore.bluegreen_v2 - InService: {('arn:aws:elasticloadbalancing:us-east-1:512686554592:targetgroup/journ-ELBv2-1OLGYF820HKCK/a577e201701478e1', 'i-09d3991bf9232ddcc'): False, ('arn:aws:elasticloadbalancing:us-east-1:512686554592:targetgroup/journ-ELBv2-1OLGYF820HKCK/a577e201701478e1', 'i-0440759a13f7f2c0d'): False}
[2021-11-12T10:55:18.653Z] INFO - buildercore.utils - done.
[2021-11-12T10:55:18.653Z] [3.91.65.159] Executing task 'single_node_work_fn'
[2021-11-12T10:55:18.653Z] [3.93.68.16] Executing task 'single_node_work_fn'
[2021-11-12T10:55:19.996Z] INFO - buildercore.bootstrap - no cfn outputs found or overwrite=True, writing /etc/cfn-info.json ...
[2021-11-12T10:55:19.996Z] INFO - buildercore.bootstrap - no cfn outputs found or overwrite=True, writing /etc/cfn-info.json ...
[2021-11-12T10:55:20.908Z] INFO - buildercore.command - uploading ? bytes to /etc/cfn-info.json
[2021-11-12T10:55:20.908Z] INFO - buildercore.command - uploading ? bytes to /etc/cfn-info.json
[2021-11-12T10:55:21.819Z] INFO - buildercore.command - uploading /ext/srv/builder/scripts/bootstrap.sh to /tmp/bootstrap.sh-20211112105521
[2021-11-12T10:55:21.819Z] INFO - buildercore.command - uploading /ext/srv/builder/scripts/bootstrap.sh to /tmp/bootstrap.sh-20211112105521


[2021-11-12T10:55:22.370Z] ERROR - buildercore.core - sudo() received nonzero return code 1 while executing!        Requested: grain_project=journal /bin/bash /tmp/bootstrap.sh-20211112105521 '3003' 'journal--prod--3' 'false' '10.0.2.135'    Executed: sudo -S -p 'sudo password:'  /bin/bash -l -c "grain_project=journal /bin/bash /tmp/bootstrap.sh-20211112105521 '3003' 'journal--prod--3' 'false' '10.0.2.135'"


[2021-11-12T10:55:22.370Z] !!! Parallel execution exception under host '3.91.65.159':
[2021-11-12T10:55:22.370Z] Process 3.91.65.159:
[2021-11-12T10:55:22.370Z] Traceback (most recent call last):
[2021-11-12T10:55:22.370Z]   File "/usr/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
[2021-11-12T10:55:22.370Z]     self.run()
[2021-11-12T10:55:22.370Z]   File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
[2021-11-12T10:55:22.370Z]     self._target(*self._args, **self._kwargs)
[2021-11-12T10:55:22.370Z]   File "/ext/srv/builder/venv/local/lib/python2.7/site-packages/fabric/tasks.py", line 241, in inner
[2021-11-12T10:55:22.370Z]     submit(task.run(*args, **kwargs))
[2021-11-12T10:55:22.370Z]   File "/ext/srv/builder/venv/local/lib/python2.7/site-packages/fabric/tasks.py", line 173, in run
[2021-11-12T10:55:22.370Z]     return self.wrapped(*args, **kwargs)
[2021-11-12T10:55:22.370Z]   File "/ext/srv/builder/venv/local/lib/python2.7/site-packages/fabric/decorators.py", line 185, in inner
[2021-11-12T10:55:22.370Z]     return func(*args, **kwargs)
[2021-11-12T10:55:22.370Z]   File "/ext/srv/builder/src/buildercore/core.py", line 398, in single_node_work_fn
[2021-11-12T10:55:22.370Z]     raise err
[2021-11-12T10:55:22.370Z] CommandException: sudo() received nonzero return code 1 while executing!
[2021-11-12T10:55:22.370Z] 
[2021-11-12T10:55:22.370Z] Requested: grain_project=journal /bin/bash /tmp/bootstrap.sh-20211112105521 '3003' 'journal--prod--3' 'false' '10.0.2.135'
```

the line:

> [2021-11-12T10:56:13.084Z] Underlying exception:
[2021-11-12T10:56:13.084Z]     sudo() received nonzero return code 1 while executing!
[2021-11-12T10:56:13.084Z]     
[2021-11-12T10:56:13.084Z]     Requested: grain_project=journal /bin/bash /tmp/bootstrap.sh-20211112105521 '3003' 'journal--prod--3' 'false' '10.0.2.135'
[2021-11-12T10:56:13.084Z]     Executed: sudo -S -p 'sudo password:'  /bin/bash -l -c "grain_project=journal /bin/bash /tmp/bootstrap.sh-20211112105521 '3003' 'journal--prod--3' 'false' '10.0.2.135'"
[2021-11-12T10:56:13.084Z] 

so, it successfully partitions the nodes into blue (2 nodes) and green (1 node) groups, starts concurrent update on blue group, with one node successful the other node fails during execution of the bootstrap script. If there is a rollback, it doesn't happen and the nodes are registered again with the ALB.

I'll take a closer look at the job and see what we're doing regarding a rollback.

--- a/incidents/2021-11-15-alb-runs-out-of-nodes.md
+++ b/incidents/2021-11-15-alb-runs-out-of-nodes.md
@@ -39,7 +39,7 @@ UTC/ACDT
 * error was expected so I ignored it
 * I check the CPU Credits on the other nodes, all very high
     - I suspect the load balancer is routing all traffic to a single node
-* I check the ALB Target Group and only one node is attached
+* I check the ALB Target Group via the console and only one node is attached
     - the one I just stopped and started.
 * I use the console to add nodes 1 and 3 back into the target group
     - the command I just ran should have done this, but if it didn't I have no time to debug why
@@ -71,13 +71,17 @@ During a blue-green deployment, the project's nodes are split into two groups an
 
 The nodes are removed from the load balancer so they don't receive traffic while updating and then updated in parallel.
 
-Something happened during the upload of the bootstrap script on one of the nodes that isn't clear and it exited with a return code of 1. The other node completed it's update successfully.
+Something happened during the upload of the bootstrap script on one of the nodes that isn't clear and it exited with a 
+return code of 1. The other node completed it's update successfully.
 
-This failure meant that the second group was not updated and left alone, presumably until the problem in the first group was rectified.
+This failure meant that the second group was not updated and left alone, presumably until the problem in the first group 
+was rectified.
 
-As there are only three nodes in the journal--prod group, we had a group of 2 nodes and a group of 1 nodes and the 2-node group was deregistered first leaving a single node to handle traffic for half of friday, the weekend and half of monday.
+As there are only three nodes in the journal--prod group, we had a group of 2 nodes and a group of 1 nodes and the 2-node 
+group was deregistered first leaving a single node to handle traffic for half of friday, the weekend and half of monday.
 
-The corrective action of adding the nodes back into the load balancer meant that the journal was operating with software in an inconsistent state.
+The corrective action of adding the nodes back into the load balancer meant that the journal was operating with software 
+in an inconsistent state.
 
 node 1: failed to update, in the same state as node 2
 node 3: successfully updated, running a different version of the journal to nodes 1 and 2.
@@ -88,9 +92,9 @@ A better solution to this incident would have been:
     - nodes would be in an inconsistent state but this is better than not serving traffic.
 2. run the [prod-journal](https://alfred.elifesciences.org/job/prod-journal/) pipeline in Jenkins once the CPU Credits and NewRelic Apdex had improved.
     - this would have run the 'Deploy to prod' step that would do the blue-green deployment again.
-    - the `build_vars` on each of the nodes would have been made consistent so that highstate would bring them all into the same state
+    - the `build_vars` on each of the nodes would have been made consistent and highstate would have made the rest of the system consistent. 
 
 ## corrective actions
 
 - [x] create a NewRelic alert for journal nodes that become suspiciously idle for a period of time
-- [ ] increase the number of journal nodes from 3 to 4 so that blue-green partitioning never leaves a group of 1 to cope with all traffic.
+- [x] increase the number of journal nodes from 3 to 4 so that blue-green partitioning never leaves a group of 1 to cope with all traffic.

--- a/incidents/2021-11-15-alb-runs-out-of-nodes.md
+++ b/incidents/2021-11-15-alb-runs-out-of-nodes.md
@@ -53,10 +53,44 @@ UTC/ACDT
 
 Added nodes back to the target group through the console.
 
+The builder approach should have been:
+
+    ./bldr deploy.load_balancer_register_all:journal--prod
+
 ## impact
 
 Maximum of 20mins of downtime and/or degraded performance.
 
+## investigation
+
+Feedback from AWS support pointed us to Cloudtrail and the load-balancer target-group deregistration events.
+
+There were events by the 'elife-alfred' user traced back to a failed `journal--prod` blue-green deployment.
+
+During a blue-green deployment, the project's nodes are split into two groups and updated one group at a time.
+
+The nodes are removed from the load balancer so they don't receive traffic while updating and then updated in parallel.
+
+Something happened during the upload of the bootstrap script on one of the nodes that isn't clear and it exited with a return code of 1. The other node completed it's update successfully.
+
+This failure meant that the second group was not updated and left alone, presumably until the problem in the first group was rectified.
+
+As there are only three nodes in the journal--prod group, we had a group of 2 nodes and a group of 1 nodes and the 2-node group was deregistered first leaving a single node to handle traffic for half of friday, the weekend and half of monday.
+
+The corrective action of adding the nodes back into the load balancer meant that the journal was operating with software in an inconsistent state.
+
+node 1: failed to update, in the same state as node 2
+node 3: successfully updated, running a different version of the journal to nodes 1 and 2.
+
+A better solution to this incident would have been:
+
+1. add the nodes back to the load balancer (as was done), either by the console or builder.
+    - nodes would be in an inconsistent state but this is better than not serving traffic.
+2. run the [prod-journal](https://alfred.elifesciences.org/job/prod-journal/) pipeline in Jenkins once the CPU Credits and NewRelic Apdex had improved.
+    - this would have run the 'Deploy to prod' step that would do the blue-green deployment again.
+    - the `build_vars` on each of the nodes would have been made consistent so that highstate would bring them all into the same state
+
 ## corrective actions
 
-...
+- [x] create a NewRelic alert for journal nodes that become suspiciously idle for a period of time
+- [ ] increase the number of journal nodes from 3 to 4 so that blue-green partitioning never leaves a group of 1 to cope with all traffic.

